### PR TITLE
[pipelined]Switch from IP to TEID UE identification

### PR DESCRIPTION
**Pull request by koolzz**
# DO NOT MERGE : pending some testing

## Summary

As part of the stats reporting rework we're switching from ip+imsi -> imsi+teid reporting

## Test Plan
unit test pass, snapshots updated

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/python/magma/pipelined/policy_converters.py
Comment (pshelar) [inline lte/gateway/python/magma/pipelined/policy_converters.py]: same function can be used for DL direction, right?

Comment (koolzz) [reply lte/gateway/python/magma/pipelined/policy_converters.py]: Yes uplink tunnel is set the same for both directions

**Comment by uri200** [comment]
Will the RuleRecord still contain the IP field, besided the TEID?

**Comment by koolzz** [comment]
> Will the RuleRecord still contain the IP field, besided the TEID?

nope only teid

File: lte/gateway/python/magma/pipelined/policy_converters.py
Comment (pshelar) [reply lte/gateway/python/magma/pipelined/policy_converters.py]: can you change the parameter name to tun_id to avoid confusion.

File: lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py
Comment (pshelar) [inline lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py]: I guess this is accidental.

**Review by pshelar — APPROVED** [review]

File: lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py
Comment (koolzz) [reply lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py]: added it to mass change test cases on rewrites like this one

File: lte/gateway/c/session_manager/test/ProtobufCreators.cpp
Comment (uri200) [inline lte/gateway/c/session_manager/test/ProtobufCreators.cpp]: Since we are not using IPs on rule records anymore, you may need to remove that parameter and cascade this to any function that creates a rule record. Plus, if you don't remove that field, you may be bypassing some tests that are checking that field. Welcome to sessiond world

**Review by uri200 — CHANGES_REQUESTED** [review]

File: lte/gateway/c/session_manager/test/ProtobufCreators.cpp
Comment (koolzz) [reply lte/gateway/c/session_manager/test/ProtobufCreators.cpp]: Lets do all sessiond review here https://github.com/magma/magma/pull/6232

Comment (koolzz) [reply lte/gateway/c/session_manager/test/ProtobufCreators.cpp]: Removing that line produces ~8 errors in test_local_enforcer. Not sure how many of those are real vs just expecting rule record to have an ip_addr. @themarwhal  could you take a look

Comment (themarwhal) [reply lte/gateway/c/session_manager/test/ProtobufCreators.cpp]: Yeah let me clean up the ip field in the other PR.

Comment (themarwhal) [reply lte/gateway/c/session_manager/test/ProtobufCreators.cpp]: @uri200 I've addressed your comment in the other PR. :)

Comment (themarwhal) [reply lte/gateway/c/session_manager/test/ProtobufCreators.cpp]: @koolzz can cherrypick again into this PR.

**Review by themarwhal — APPROVED** [review]
All sessiond changes have been reviewed in the other PR

File: lte/gateway/c/session_manager/test/ProtobufCreators.cpp
Comment (uri200) [reply lte/gateway/c/session_manager/test/ProtobufCreators.cpp]: thanks. Sorry my initial comment was missing `you may need to remove that parameter` not to sound rude 😅

**Review by uri200 — APPROVED** [review]

**Comment by ekowtaylor** [comment]
@themarwhal , @uri200 is this ready to be merged?

**Comment by themarwhal** [comment]
> @themarwhal , @uri200 is this ready to be merged?

No :( We are still testing this PR.

**Comment by ekowtaylor** [comment]
Once this PR has been approved and passing all CI checks, use the ready2merge label to indicate the maintainers can merge your PR.

**Comment by ekowtaylor** [comment]
@themarwhal what's the latest? Can this be closed now?

**Comment by themarwhal** [comment]
> @themarwhal what's the latest? Can this be closed now?

cc @koolzz

**Comment by koolzz** [comment]
> > @themarwhal what's the latest? Can this be closed now?
> 
> cc @koolzz

Will schedule a meeting to discuss

**Comment by koolzz** [comment]
In the middle of resolving some nasty merge conflicts/don't review

**Comment by ekowtaylor** [comment]
@koolzz updates on merge conflicts?

**Comment by ekowtaylor** [comment]
@koolzz,  this PR has now gone stale. To maintain a healthy codebase, it will be automatically closed unless there are plans to rebase and merge by close of week

**Comment by ekowtaylor** [comment]
To be reopened when ready to move forward

**Pull request closed without merging**
